openmetadata-ui/src/main/resources/ui/cypress/common/common.js

```diff
@@ -52,25 +58,45 @@ export const handleIngestionRetry = (type, testIngestionButton, count = 0) => {
     testIngestionsTab();
     retryCount++;
     // the latest run should be success
-    cy.get('.tableBody-row > :nth-child(4)').then(($ingestionStatus) => {
-      if (
-        ($ingestionStatus.text() === 'Running' ||
-          $ingestionStatus.text() === 'Queued') &&
-        retryCount <= retryTimes
-      ) {
-        // retry after waiting for 20 seconds
-        cy.wait(20000);
-        cy.reload();
-        checkSuccessState();
-      } else {
-        cy.get('.tableBody-row > :nth-child(4)').should('have.text', 'Success');
+    cy.get(`.tableBody > :nth-child(${rowIndex}) > :nth-child(4)`).then(
+      ($ingestionStatus) => {
+        if (
+          ($ingestionStatus.text() === 'Running' ||
+            $ingestionStatus.text() === 'Queued') &&
+          retryCount <= retryTimes
+        ) {
+          // retry after waiting for 20 seconds
+          cy.wait(20000);
```

Comment:
Can we check if we can use waitFor here instead hardcoded timer?

Reply:
Hello, This is to wait for run ingestion on the airflow side, as we don't make any API call from UI, it is done from the ingestion side, and we only get status once its a success, so we have to hardcoded the timer!